### PR TITLE
#160451557 Reset Password Functionality

### PR DESCRIPTION
**Pull request by andela-phemmy**
#### What does this PR do?
     This PR implements Reset/Forgot password functionality
#### Description of Task to be completed?
      1. Configure the resetReducer
      2. Configure the reset action and add action type
      3. Add containers and components
      4. Test reducers, actions and components
#### How should this be manually tested?
      On localhost
        1. start the app on port `8080`
        2. Navigate to sign in
        3. Click on forgot password
        4. input your password 
        5. An email will be sent to you, click on reset password
        6. input password and confirm password
        7. proceed to reset your password
#### Any background context you want to provide?
        N/A
### What are the relevant pivotal tracker stories?
       #160451557
#### Screenshots (if appropriate)
screen shot 2018-11-13 at 1 13 36 pm
screen shot 2018-11-13 at 1 13 49 pm
screen shot 2018-11-13 at 1 13 58 pm
screen shot 2018-11-13 at 1 14 16 pm

#### Questions:
        N/A

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -59,4 +59,4 @@
           }
       }]
     }
-  }
+  }
```
Comment (e0works) [inline .eslintrc]: You forgot to add a new line at the end of this file, please update the file

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ {
+     "singleQuote": true,
+     "trailingComma": "es5",
+     "jsxSingleQuote": true,
+     "jsxBracketSameLine": true
+ }
```
Comment (e0works) [inline .prettierrc]: Please, add a new line here also

File: src/components/ForgotPassword/ForgotPassword.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
+// react libraries
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
```
Comment (e0works) [inline src/components/ForgotPassword/ForgotPassword.jsx:5]: Please, comment appropriately
prop-types and redux are not part of React library
it should read

```js
        // react libraries
        import React, { Component } from 'react';

        // tthird-party libraries
        import PropTypes from 'prop-types';
	import { connect } from 'react-redux';
```

File: src/components/ForgotPassword/ForgotPasswordForm.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
+// react libraries
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
```
Comment (e0works) [inline src/components/ForgotPassword/ForgotPasswordForm.jsx]: Here also, give a space between lines 2 and 3 and comment the imports properly

```js
// third-party libraries
import PropTypes from 'prop-types';
```

File: src/components/ResetPassword/ResetPasswordForm.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
+// react and react dependent library
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
```
Comment (e0works) [inline src/components/ResetPassword/ResetPasswordForm.jsx:5]: separate  ***prop-types*** from **react** library

File: src/components/login/LoginForm.jsx [hunk 6]
```diff
@@ -2,9 +2,8 @@
 import React, { Component, Fragment } from 'react';
```
Comment (e0works) [inline src/components/login/LoginForm.jsx:2]: comment on the imports here

File: src/components/reusables/Loader.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
```
Comment (e0works) [inline src/components/reusables/Loader.jsx:2]: Add comment on the import statements here

File: src/index.html [hunk 8]
```diff
@@ -4,9 +4,12 @@
   <meta charset="UTF-8">
```
Comment (e0works) [inline src/index.html]: I don't think we are making use of this file, so it should be deleted

**Review by e0works — CHANGES_REQUESTED** [review]

File: .eslintrc [hunk 9]
```diff
@@ -60,3 +60,4 @@
       }]
     }
   }
+
```
Comment (Mrshinezee) [inline .eslintrc:63]: add new line

File: .prettierrc [hunk 10]
```diff
@@ -0,0 +1,7 @@
...
+
```
Comment (Mrshinezee) [inline .prettierrc:7]: add a new line

**Review by Mrshinezee — CHANGES_REQUESTED** [review]

**Review by e0works — APPROVED** [review]

**Review by Mrshinezee — APPROVED** [review]
LGTM

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]
@andela-phemmy 
Rebase.

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**